xarray/backends/zarr.py

```diff
@@ -1066,6 +1066,8 @@ def _open_existing_array(self, *, name) -> ZarrArray:
         else:
             zarr_array = self.zarr_group[name]
 
+        if TYPE_CHECKING:
```

Comment:
Not sure what the error here was, but simpler use `cast(ZarrArray, zarr_array)`.